src/Microsoft.AspNet.Mvc.ViewFeatures/DependencyInjection/MvcViewFeaturesMvcCoreBuilderExtensions.cs

```diff
@@ -119,9 +119,11 @@ internal static void AddViewServices(IServiceCollection services)
             //
             // View Components
             //
+            services.TryAddSingleton<IViewComponentActivator, DefaultViewComponentActivator>();
+            
             // These do caching so they should stay singleton
             services.TryAddSingleton<IViewComponentSelector, DefaultViewComponentSelector>();
-            services.TryAddSingleton<IViewComponentActivator, DefaultViewComponentActivator>();
+            services.TryAddSingleton<IViewComponentFactory, DefaultViewComponentFactory>();
```

Comment:
The activator is the one that belongs with this comment. Personally I'm game for just deleting the comments. Everything for the most part in the framework is singleton by default.